app/models/spree/line_item.rb

```diff
@@ -122,6 +123,15 @@ def copy_tax_category
       self.tax_category = variant.product.tax_category
     end
 
+    def copy_dimensions
+      return unless variant
+
+      self.weight ||= variant.product.variant_unit == "weight" ? variant.unit_value / variant.product.variant_unit_scale : variant.weight
```

Comment:
This is a very long line. Can we extract that to a method?

Reply:
Yes!

Reply:
Much better.